tools/updater/main.c

```diff
@@ -707,407 +669,345 @@ static k4a_result_t ensure_firmware_open(updater_command_info_t *command_info)
-            device = NULL;
-            continue;
+            print_firmware_build_config(command_info->current_version.firmware_build);
+            print_firmware_signature_type(command_info->current_version.firmware_signature, true);
+            printf("\n");
         }
-
-        if (K4A_BUFFER_RESULT_SUCCEEDED != firmware_get_device_serialnum(device, serial_number, &serial_number_length))
+        else
         {
-            printf("ERROR: %d: Failed to get serial number\n", deviceIndex);
-            free(serial_number);
-            serial_number = NULL;
-            firmware_destroy(device);
-            device = NULL;
-            continue;
+            printf("ERROR: Failed to get current versions\n\n");
+            return K4A_RESULT_FAILED;
         }
-
-        printf("%d: Device \"%s\"\n", deviceIndex, serial_number);
-
-        free(serial_number);
-        serial_number = NULL;
-        firmware_destroy(device);
-        device = NULL;
-    }
-}
-
-static k4a_result_t command_query_device(updater_command_info_t *command_info)
-{
-    k4a_result_t result = ensure_firmware_open(command_info);
-    if (!K4A_SUCCEEDED(result))
-    {
-        return result;
-    }
-
-    result = print_device_serialnum(command_info);
-    if (!K4A_SUCCEEDED(result))
-    {
-        return result;
-    }
-
-    result = firmware_get_device_version(command_info->firmware_handle, &command_info->current_version);
-    if (K4A_SUCCEEDED(result))
-    {
-        printf("Current Firmware Versions:\n");
-        printf("  RGB camera firmware:      %d.%d.%d\n",
-               command_info->current_version.rgb.major,
-               command_info->current_version.rgb.minor,
-               command_info->current_version.rgb.iteration);
-        printf("  Depth camera firmware:    %d.%d.%d\n",
-               command_info->current_version.depth.major,
-               command_info->current_version.depth.minor,
-               command_info->current_version.depth.iteration);
-        printf("  Depth config file:        %d.%d\n",
-               command_info->current_version.depth_sensor.major,
-               command_info->current_version.depth_sensor.minor);
-        printf("  Audio firmware:           %d.%d.%d\n",
-               command_info->current_version.audio.major,
-               command_info->current_version.audio.minor,
-               command_info->current_version.audio.iteration);
-
-        print_firmware_build_config(command_info->current_version.firmware_build);
-        print_firmware_signature_type(command_info->current_version.firmware_signature, true);
-        printf("\n");
-    }
-    else
-    {
-        printf("ERROR: Failed to get current versions\n\n");
-        return K4A_RESULT_FAILED;
     }
 
     return K4A_RESULT_SUCCEEDED;
 }
 
-static k4a_result_t command_inspect_firmware(updater_command_info_t *command_info)
+static k4a_result_t command_inspect_firmware(char *firmware_path, firmware_package_info_t *firmware_info)
```

Comment:
Since it is a little confusing, could we rename this to "command_load_and_inspect_firmware"?